src/query/storage/m3/storage.go

```diff
@@ -327,17 +311,22 @@ func (s *m3storage) fetchCompressed(
 				)
 			}
 
+			if err != nil {
+				result.ReportError(err)
+				return
```

Comment:
Now that this is possible to receive an error, before returning at the bottom of top level function `fetchCompressed` here should we do:
```go
if err := result.Err(); err != nil {
  return nil, err
}
return result, nil
```

Seems more sane than letting someone run into the `iter.Err` later perhaps?